Frameworks/SystemConfiguration/SCNetworkReachability.mm

```diff
@@ -134,15 +134,19 @@ + (void)_testAllReachabilityObjects {
 
 + (void)_checkGlobalReachability {
     WNCConnectionProfile* internetConnectionProfile = [WNCNetworkInformation getInternetConnectionProfile];
-    if (internetConnectionProfile &&
-        [internetConnectionProfile getNetworkConnectivityLevel] == WNCNetworkConnectivityLevelInternetAccess) {
-        SCNetworkReachabilityFlags newFlags = kSCNetworkReachabilityFlagsReachable;
+    @try {
+        if (internetConnectionProfile &&
+            [internetConnectionProfile getNetworkConnectivityLevel] == WNCNetworkConnectivityLevelInternetAccess) {
+            SCNetworkReachabilityFlags newFlags = kSCNetworkReachabilityFlagsReachable;
 
-        if (internetConnectionProfile.isWwanConnectionProfile) {
-            newFlags |= kSCNetworkReachabilityFlagsIsWWAN;
+            if (internetConnectionProfile.isWwanConnectionProfile) {
+                newFlags |= kSCNetworkReachabilityFlagsIsWWAN;
+            }
+            _globalReachableFlags = newFlags;
+        } else {
+            _globalReachableFlags = 0;
         }
-        _globalReachableFlags = newFlags;
-    } else {
+    } @catch (NSException* exception) {
```

Comment:
Do we have a compiler bug/task that's tracking this? +Wes

---
In reply to: [97645904](https://github.com/Microsoft/WinObjC/pull/1795#discussion_r97645904) [](ancestors = 97645904)